GitTfs.Vs2017/TfsHelper.Vs2017.cs

```diff
@@ -0,0 +1,129 @@
+﻿using System;
+using System.IO;
+using System.Linq;
+using System.Runtime.InteropServices;
+using Microsoft.TeamFoundation.Build.Client;
+using Microsoft.TeamFoundation.Client;
+using Microsoft.TeamFoundation.Server;
+using Microsoft.TeamFoundation.VersionControl.Client;
+using Microsoft.VisualStudio.Setup.Configuration;
+using Sep.Git.Tfs.Core.TfsInterop;
+using Sep.Git.Tfs.VsCommon;
+using StructureMap;
+
+namespace Sep.Git.Tfs.Vs2017
+{
+    public class TfsHelper : TfsHelperBase
+    {
+        private const int REGDB_E_CLASSNOTREG = unchecked((int)0x80040154);
+
+        private const string tfsExtensionsFolder =
+            @"Common7\IDE\CommonExtensions\Microsoft\TeamFoundation\Team Explorer";
+
+        protected string TfsVersionString { get { return "15.0"; } }
+
+        public TfsHelper(TfsApiBridge bridge, IContainer container)
+            : base(bridge, container)
+        { }
+
+        protected override bool HasWorkItems(Changeset changeset)
+        {
+            return Retry.Do(() => changeset.AssociatedWorkItems.Length > 0);
+        }
+
+        protected override string GetVsInstallDir()
+        {
+            try
+            {
+                string visualStudioInstallationPath = null;
+
+                var setupConfiguration = (ISetupConfiguration2)GetSetupConfiguration();
+                var instancesEnumerator = setupConfiguration.EnumAllInstances();
+
+                int fetched;
+                var instances = new ISetupInstance[1];
+                do
+                {
+                    instancesEnumerator.Next(1, instances, out fetched);
+                    if (fetched > 0)
+                    {
+                        var instance = (ISetupInstance2)instances[0];
+                        var state = instance.GetState();
+                        if ((state & InstanceState.Local) == InstanceState.Local)
+                        {
+                            visualStudioInstallationPath = instance.GetInstallationPath();
+                        }
+                    }
+                }
+                while (fetched > 0 && !string.IsNullOrEmpty(visualStudioInstallationPath));
+
+                return visualStudioInstallationPath;
+            }
+            catch (Exception ex)
+            {
+                throw ex;
+            }
+        }
+
+        protected override IBuildDetail GetSpecificBuildFromQueuedBuild(IQueuedBuild queuedBuild, string shelvesetName)
+        {
+            var build = queuedBuild.Builds.FirstOrDefault(b => b.ShelvesetName == shelvesetName);
+            return build != null ? build : queuedBuild.Build;
+        }
+
+#pragma warning disable 618
```

Comment:
I know it says I added it, but this code was actually copied from another file which also contained this warning suppression (from the GitTfs.Vs2015 project). I merely used the GitTfs.Vs2015 project as a template, and then customized it for the required changes to support Visual Studio 2017.